src/Plugin/AddPathPlugin.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Add base path to the request URI. Useful for base API URLs like http://domain.com/api.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class AddPathPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $host;
+
+    /**
+     * @param UriInterface $host
+     */
+    public function __construct(UriInterface $host)
+    {
+        if (rtrim($host->getPath(), '/') === '') {
+            throw new \LogicException('Host path cannot be empty');
+        }
+
+        $this->host = $host;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
+    {
+        $request = $request->withUri($request->getUri()
+            ->withPath(str_replace('//', '/', $this->host->getPath().$request->getUri()->getPath()))
```

Comment:
If we throw an exception like asked in https://github.com/php-http/client-common/pull/48#discussion_r83802537, we don't need to rtrim anything IMHO.